### PR TITLE
Add SingleFile support for saving transcript whole local

### PR DESCRIPTION
**Pull request by saymoon**
- Add SingleFile support for saving transcript whole local
- Fix bug

### CODE REVIEW

File: vistopia/visitor.py [hunk 1]
```diff
@@ -114,19 +114,64 @@ def save_transcript(self, id: int, episodes: Optional[set] = None):
...
-
-        track = EasyID3(fname)
-        track["title"] = article_info["title"]
-        track["album"] = series_info["title"]
-        track["artist"] = series_info["author"]
-        track["tracknumber"] = article_info["sort_number"]
-        # track["tracksort"] = article_info["sort_number"]
-        track["website"] = article_info["content_url"]
-        track.save()
+        from mutagen.id3._util import ID3NoHeaderError
```
Comment (chazeon) [inline vistopia/visitor.py]: The `ID3NoHeaderError` can be imported from `mutagen.id3` directly:
* https://github.com/quodlibet/mutagen/blob/f95d3ae19e25e3f0a91061566551843d799317c5/mutagen/id3/__init__.py#L37
* https://github.com/quodlibet/mutagen/issues/327

File: vistopia/visitor.py [hunk 2]
```diff
@@ -114,19 +114,64 @@ def save_transcript(self, id: int, episodes: Optional[set] = None):
...
+
+        try:
+            track = EasyID3(fname)
+        except ID3NoHeaderError:
+            # No ID3 tag found, creating a new ID3 tag
+            track = EasyID3()
+            track.save(fname)
+            track = EasyID3(fname)
```
Comment (chazeon) [inline vistopia/visitor.py]: Save and reopen seems unnecessary:
https://github.com/quodlibet/mutagen/issues/327#issuecomment-339316014

Is this track publicly available or is a paid episode?

File: vistopia/main.py [hunk 3]
```diff
@@ -130,19 +131,33 @@ def save_show(ctx, **argv):
 @main.command("save-transcript")
 @click.option("--id", type=click.INT, required=True)
 @click.option("--episode-id", help="Episode ID in the form '1-3,4,8'")
+@click.option("--single-file-exec-path", type=click.Path(),
+              help="Path to the single-file CLI tool")
+@click.option("--cookie-file-path", type=click.Path(),
+              help="Path to the browser cookie file")
```
Comment (chazeon) [inline vistopia/main.py]: Please mention in the help that it is used in combination with the single-file option only.

**Review by chazeon — CHANGES_REQUESTED** [review]

File: vistopia/visitor.py (hunk 2 above)
Comment (saymoon) [reply vistopia/visitor.py]: This issue can be reproduced in show 163 episode 81, it's a paid episode.

**Pull request merged**